### DISCUSSION TITLE
WezTerm stopped loading my configuration file unless I specify it with --config-file

### DISCUSSION THREAD

**Discussion by leggettc18** [wezterm/wezterm#5856] [Q&A]
### What Operating System(s) are you running on?

Linux X11

### Which Wayland compositor or X11 Window manager(s) are you using?

Gnome (Specifically Pop OS)

### WezTerm version

wezterm 20240203-110809-5046fc22

### Ask your question!

All of a sudden today my WezTerm just stopped loading (or maybe stopped finding?) my configuration file. It is correctly located in ~/.config/wezterm/wezterm.lua (or a symlink to it is there, actual file is in a dotfiles repo) and if I run wezterm from another terminal with the --config-file option set, it loads my config perfectly fine. I'm certain there must be something environmental causing it but I have no idea what it could be. Any ideas?

**Comment by leggettc18** [comment]
For now I uninstalled the Flatpak and installed via the apt repo instead and that seems to have worked.

**Comment by wez** [comment]
There are certain kinds of config parsing error that will prevent the automatic configuration reloading from working, until you fix them and manually reload the config.
You can use your [ReloadConfiguration](https://wezfurlong.org/wezterm/config/lua/keyassignment/ReloadConfiguration.html) key assignment (default CTRL-SHIFT-R) or the Command Palette (CTRL-SHIFT-P, then type "reload" and hit enter) to manually reload, and if you have resolved the error, reloading should continue to work automatically from that point.

**Reply by leggettc18** [reply]
That definitely isn't it. Like I said if I manually specified the config file via the `--config-file` cli switch it loads just fine. I also just installed the apt version and it also loaded just fine at that point. My best theory at this point is something happened with my flatpak configuration.

**Reply by IvayloSt99** [reply]
I have the exact same issue on EndeavourOS with KDE on Wayland. Version 20250703-070941-c7f4b081. It doesn't make a difference whether I install wezterm-git from AUR or the package in the official repository and reloading doesn't change anything either. The same config works just fine on Windows and also on Linux if I run it from a terminal with --config-file specified.

**Reply by pfych** [reply]
I'm also experiencing this issue on EndeavourOS with KDE on Wayland.

**Wezterm Version:** `wezterm 20240203-110809-5046fc22`
**Linux:** `6.16.7-arch1-1`

Config doesn't load automatically anymore, but `--config-file` works.

**Reply by bew** [reply]
@pfych did you tried restarting your computer?
2 people had the same problem recently in #7233 where a full reboot fixed their issue 🙄

**Reply by pfych** [reply]
I’ve just done 3 full reboots and am still having issues with my config loading.

Copying the file to the home folder (`~/.wezterm.lua`) works but is not ideal as I’d rather it live in `~/.config/wezterm/wezterm.lua`.

**Comment by nasxisbest** [comment]
Stopped loading config from ~/.config after recent upgrade

**Reply by nasxisbest** [reply]
UPD: It's all `/terminal-setup` from Claude Code lmaoo, just delete the `~/.wezterm.lua` if you have one at `~/.config/wezterm/wezterm.lua`

**Reply by Wangjianhui2003** [reply]
great help! thanks

**Comment by BigJayToDaIzo** [comment]
updating today broke my config too.  --config-file works and moving my config out of xdg_config and into ~/.wezterm.lua works. Neither option is desirable.  I want my config files all within my .config directory with the rest of them. This makes managing my configuration repository troublesome. Weirdly, once I delete ~/.wezterm.lua (I copied my config from my original location) it properly loads from xdg_config properly again?

**Comment by amsesk** [comment]
I debugged this a ton as well. It turns out that I had an bare-bones config file at `$HOME/.wezterm.lua` that I did not realize that I had. Deleting it fixed the issue for me. 

Perhaps the priority of the different file locations has been changed?

**Reply by vishes-shell** [reply]
Thank you! I had the same thing, i don't know where it came from...
This worked for me!🙏

**Reply by lborgav** [reply]
This is the answer! Tks

**Reply by mdubbelman** [reply]
This was the solution for me too! 
The new config file in `~/.wezterm.lua` was created by Claude Code after I used the `/terminal-setup` command.

**Reply by bew** [reply]
> This was the solution for me too! The new config file in `~/.wezterm.lua` was created by Claude Code after I used the `/terminal-setup` command.

Interesting, I tried to find where this command is implemented and realized that [claude-code rs NOT open-source](https://github.com/anthropics/claude-code/issues/1789) 😢

I'll have to make an issue later on (or someone else, feel free!)

**Reply by ealves-pt** [reply]
Same issue here and claude code was the culprit.

**Reply by khaled-maruf** [reply]
same issue, It seems approving ClaudeCode to work on `SHIFT+ENTER` to add new line in the prompt made it to create this config file at `~/.wezterm.lua`, darn....

**Reply by DoKoB0512** [reply]
This is very annoying to debug. Luckily we can get the solution from these comments.
